pkg/controlplane/controller/leaderelection/run_with_leaderelection.go

```diff
@@ -50,7 +50,9 @@ func RunWithLeaderElection(ctx context.Context, config *rest.Config, newRunnerFn
 			run(ctx, 1)
 		},
 		OnStoppedLeading: func() {
-			cancel()
+			if cancel != nil {
```

Comment:
Is it better to make the changes in the Run() method? Should we execute the OnStoppedLeading method after the OnStartedLeading method is called?
```go
func (le *LeaderElector) Run(ctx context.Context) {
	defer runtime.HandleCrash()

	// Ensure OnStoppedLeading is only called if OnStartedLeading has been called
	var startedLeading bool

	ctx, cancel := context.WithCancel(ctx)
	defer func() {
		if startedLeading {
			if le.config.Callbacks.OnStoppedLeading != nil {
				le.config.Callbacks.OnStoppedLeading()
			}
		}
		cancel()
	}()

	if !le.acquire(ctx) {
		return
	}

	go func() {
		le.config.Callbacks.OnStartedLeading(ctx)
		startedLeading = true
	}()

	le.renew(ctx)
}
```
```

Reply:
@Jefftree

Reply:
yeap, seems much better 👍